composites/Plugin/SnippetEditor/components/ReplacementVariableEditor.js

```diff
@@ -0,0 +1,167 @@
+import React from "react";
+import { EditorState, convertToRaw, convertFromRaw } from "draft-js";
+import Editor from "draft-js-plugins-editor";
+import createMentionPlugin, { defaultSuggestionsFilter } from "draft-js-mention-plugin";
+import { serializeEditor, unserializeEditor } from "../serialization";
+import flow from "lodash/flow";
+import PropTypes from "prop-types";
+import { replacementVariablesShape } from "../constants";
+
+/**
+ * Creates a DraftJS editor state from a string.
+ *
+ * @param {string} content The content to turn into editor state.
+ *
+ * @returns {EditorState} The editor state.
+ */
+const createEditorState = flow( [
+	unserializeEditor,
+	convertFromRaw,
+	EditorState.createWithContent,
+] );
+
+/**
+ * Serializes the DraftJS editor state into a string.
+ *
+ * @param {EditorState} The current editor state.
+ *
+ * @returns {string} The serialized editor state.
+ */
+const serializeEditorState = flow( [
+	convertToRaw,
+	serializeEditor,
+] );
+
+class ReplaceVarEditor extends React.Component {
+	/**
+	 * Constructs the replacement variable editor for use.
+	 *
+	 * @param {Object} props The props to instantiate this editor with.
+	 *
+	 * @returns {void}
+	 */
+	constructor( props ) {
+		super( props );
+
+		const { content } = this.props;
+
+		this.state = {
+			editorState: createEditorState( content ),
+			replacementVariables: props.replacementVariables,
+		};
+
+		this._serializedContent = content;
+
+		this.onChange = this.onChange.bind( this );
+		this.onSearchChange = this.onSearchChange.bind( this );
+
+		this.mentionsPlugin = createMentionPlugin( {
```

Comment:
Given that the mentions plugin is used for replacement variables it would probably be helpful to add a comment here why we're using the mentions plugin.